src/app/cli/src/coda_worker_testnet.ml

```diff
@@ -157,84 +178,64 @@ let start_prefix_check log workers events testnet ~acceptable_delay =
     (Linear_pipe.iter events ~f:(function `Transition (i, (prev, curr)) ->
          Linear_pipe.write all_transitions_w (prev, curr, i) ))
 
-let start_payment_check log events payments workers testnet ~acceptable_delay =
-  let block_counts = Array.init (List.length workers) ~f:(fun _ -> 0) in
-  let active_accounts = ref [] in
-  let get_balances pk =
-    Deferred.List.all
-      (List.mapi workers ~f:(fun i w -> Api.get_balance testnet i pk))
-  in
-  let add_to_active_accounts pk =
-    match List.findi !active_accounts ~f:(fun i (apk, _, _) -> apk = pk) with
-    | None ->
-        let%map balances = get_balances pk in
-        let send_block_counts = Array.to_list block_counts in
-        assert (List.length balances = List.length send_block_counts) ;
-        active_accounts :=
-          (pk, send_block_counts, balances) :: !active_accounts
-    | Some (i, a) -> return ()
-  in
-  let check_active_accounts () =
-    let%map new_aa =
-      Deferred.List.filter !active_accounts
-        ~f:(fun (pk, send_block_counts, send_balances) ->
-          let%bind balances = get_balances pk in
-          let current_block_counts = Array.to_list block_counts in
-          let%map dones =
-            Deferred.List.all
-              (List.init (List.length send_block_counts) ~f:(fun i ->
-                   let balance = List.nth_exn balances i in
-                   let send_block_count = List.nth_exn send_block_counts i in
-                   let current_block_count =
-                     List.nth_exn current_block_counts i
-                   in
-                   let send_balance = List.nth_exn send_balances i in
-                   if Option.is_none balance || Option.is_none send_balance
-                   then return true
-                   else if balance <> send_balance then return true
-                   else
-                     let diff = current_block_count - send_block_count in
-                     Logger.warn log
-                       !"%d balance not yet updated %{sexp: \
-                         Currency.Balance.t option option} %d"
-                       i balance diff ;
-                     if diff >= 7 then (
-                       Logger.fatal log "balance took too long to update" ;
-                       ignore (exit 1) ) ;
-                     return false ))
-          in
-          let all_done = List.for_all dones ~f:Fn.id in
-          if all_done then false else true )
+let start_payment_check log best_tip_diff_pipe payment_pipe workers testnet
+    ~acceptable_delay =
+  let best_tip_lengths = Array.init (List.length workers) ~f:(Fn.const 0) in
+  let active_user_commands = ref [] in
+  let add_to_active_user_commands cmd =
```

Comment:
In the old version items were removed from `active_accounts` once the payments was done processing. Please keep that behavior.